modules/core/test/v2/unit/coins/stx.ts

```diff
@@ -0,0 +1,58 @@
+import * as Promise from 'bluebird';
+import { Stx } from '../../../../src/v2/coins/';
+
+const co = Promise.coroutine;
+import { TestBitGo } from '../../../lib/test_bitgo';
+
+describe('STX:', function() {
+  let bitgo;
+  let basecoin;
+
+  before(function() {
+    bitgo = new TestBitGo({ env: 'mock' });
+    bitgo.initializeTestVars();
+    basecoin = bitgo.coin('tstx');
+  });
+
+  it('should instantiate the coin', function() {
+    const basecoin = bitgo.coin('stx');
+    basecoin.should.be.an.instanceof(Stx);
+  });
+
+
+  it('should check valid addresses', co(function *() {
+    const badAddresses = [
+      '',
+      null,
+      'abc',
+      'SP244HYPYAT2BB2QE513NSP81HTMYWBJP02HPGK6',
+      'ST1T758K6T2YRKG9Q0TJ16B6FP5QQREWZSESRS0PY'
+    ];
+    const goodAddresses = [
+      'STB44HYPYAT2BB2QE513NSP81HTMYWBJP02HPGK6',
+      'ST11NJTTKGVT6D1HY4NJRVQWMQM7TVAR091EJ8P2Y',
+      'SP2T758K6T2YRKG9Q0TJ16B6FP5QQREWZSESRS0PY',
+      'SP2T758K6T2YRKG9Q0TJ16B6FP5QQREWZSESRS0PY',
+    ];
+
+    badAddresses.map(addr => { basecoin.isValidAddress(addr).should.equal(false); });
+    goodAddresses.map(addr => { basecoin.isValidAddress(addr).should.equal(true); });
+  }));
+
+  describe('Keypairs:', () => {
```

Comment:
Is this left over from the keypair generation task?

Reply:
I looked at a few other coins' test code, and copied several of the methods from there.

While the test here might not be necessary, it did expose a flaw -- `modules/core/test/v2/unit/keychains.ts` expects to check secp256k1 coins for *extended* key support, using `getExtendedKeys()` in account-lib's `baseCoin/secp256k1ExtendedKeyPair.ts`.  I modified `core/src/v2/coins/stx.ts` to return extended keys in `generateKeyPair()`.  That led to another problem -- `isValidPub` doesn't support extended keys.  The right place to make a change there is in account-lib.